src/mqt/bench/benchmarks/ae.py

```diff
@@ -4,66 +4,66 @@
 
 import numpy as np
 from qiskit import QuantumCircuit
-from qiskit_algorithms import AmplitudeEstimation, EstimationProblem
 
 
-def create_circuit(num_qubits: int) -> QuantumCircuit:
+def create_circuit(num_qubits: int, probability: float = 0.2) -> QuantumCircuit:
     """Returns a quantum circuit implementing Quantum Amplitude Estimation.
 
     Arguments:
-        num_qubits: number of qubits of the returned quantum circuit
-    """
-    ae = AmplitudeEstimation(
-        num_eval_qubits=num_qubits - 1,  # -1 because of the to be estimated qubit
-    )
-    problem = get_estimation_problem()
-
-    qc = ae.construct_circuit(problem)
-    qc.name = "ae"
-    qc.measure_all()
+        num_qubits: Total number of qubits, including evaluation and target qubits.
+        probability: Probability of the "good" state.
 
-    return qc
+    Returns:
+        QuantumCircuit: The constructed amplitude estimation circuit.
+    """
+    if num_qubits < 2:
+        msg = "Number of qubits must be at least 2 (1 evaluation + 1 target)."
+        raise ValueError(msg)
 
+    num_eval_qubits = num_qubits - 1  # Number of evaluation qubits
+    qc = QuantumCircuit(num_qubits, num_eval_qubits)
 
-class BernoulliQ(QuantumCircuit):  # type: ignore[misc]
-    """A circuit representing the Bernoulli Q operator."""
+    # Define the Bernoulli A operator
+    theta_p = 2 * np.arcsin(np.sqrt(probability))
+    a = QuantumCircuit(1)
+    a.ry(theta_p, 0)
 
-    def __init__(self, probability: float) -> None:
-        """Initialize the Bernoulli Q operator."""
-        super().__init__(1)  # circuit on 1 qubit
+    # Apply Hadamard gates to the evaluation qubits
+    qc.h(range(num_eval_qubits))
 
-        self._theta_p = 2 * np.arcsin(np.sqrt(probability))
-        self.ry(2 * self._theta_p, 0)
```

Comment:
Here, there is a factor of 2 in there and that one is removed in the update. Could you briefly check whether this makes sense?